From 8fda84f2b3d84e8c71a69d646f48ad01e1178a8a Mon Sep 17 00:00:00 2001
From: Martin Schiller <ms@dev.tdt.de>
Date: Tue, 18 Jan 2022 11:32:54 +0100
Subject: [PATCH] MIPS: add some ifndefs around global variables.

This variables need to be changed for the Intel MIPS interAptiv SoCs
made by Intel.

Signed-off-by: Martin Schiller <ms@dev.tdt.de>
---
 arch/mips/include/asm/addrspace.h           | 40 +++++++++++++++++++++
 arch/mips/include/asm/mach-generic/spaces.h |  2 ++
 arch/mips/include/asm/pgtable-32.h          |  2 ++
 3 files changed, 44 insertions(+)

diff --git a/arch/mips/include/asm/addrspace.h b/arch/mips/include/asm/addrspace.h
index 59a48c60a065..91050ccb676e 100644
--- a/arch/mips/include/asm/addrspace.h
+++ b/arch/mips/include/asm/addrspace.h
@@ -45,12 +45,16 @@
 /*
  * Returns the kernel segment base of a given address
  */
+#ifndef KSEGX
 #define KSEGX(a)		((_ACAST32_(a)) & _ACAST32_(0xe0000000))
+#endif
 
 /*
  * Returns the physical address of a CKSEGx / XKPHYS address
  */
+#ifndef CPHYSADDR
 #define CPHYSADDR(a)		((_ACAST32_(a)) & 0x1fffffff)
+#endif
 #define XPHYSADDR(a)		((_ACAST64_(a)) &			\
 				 _CONST64_(0x0000ffffffffffff))
 
@@ -77,34 +81,70 @@
 
 #else
 
+#ifndef CKSEG0ADDR
 #define CKSEG0ADDR(a)		(CPHYSADDR(a) | KSEG0)
+#endif
+#ifndef CKSEG1ADDR
 #define CKSEG1ADDR(a)		(CPHYSADDR(a) | KSEG1)
+#endif
+#ifndef CKSEG2ADDR
 #define CKSEG2ADDR(a)		(CPHYSADDR(a) | KSEG2)
+#endif
+#ifndef CKSEG3ADDR
 #define CKSEG3ADDR(a)		(CPHYSADDR(a) | KSEG3)
+#endif
 
 /*
  * Map an address to a certain kernel segment
  */
+#ifndef KSEG0ADDR
 #define KSEG0ADDR(a)		(CPHYSADDR(a) | KSEG0)
+#endif
+#ifndef KSEG1ADDR
 #define KSEG1ADDR(a)		(CPHYSADDR(a) | KSEG1)
+#endif
+#ifndef KSEG2ADDR
 #define KSEG2ADDR(a)		(CPHYSADDR(a) | KSEG2)
+#endif
+#ifndef KSEG3ADDR
 #define KSEG3ADDR(a)		(CPHYSADDR(a) | KSEG3)
+#endif
 
 /*
  * Memory segments (32bit kernel mode addresses)
  * These are the traditional names used in the 32-bit universe.
  */
+#ifndef KUSEG
 #define KUSEG			0x00000000
+#endif
+#ifndef KSEG0
 #define KSEG0			0x80000000
+#endif
+#ifndef KSEG1
 #define KSEG1			0xa0000000
+#endif
+#ifndef KSEG2
 #define KSEG2			0xc0000000
+#endif
+#ifndef KSEG3
 #define KSEG3			0xe0000000
+#endif
 
+#ifndef CKUSEG
 #define CKUSEG			0x00000000
+#endif
+#ifndef CKSEG0
 #define CKSEG0			0x80000000
+#endif
+#ifndef CKSEG1
 #define CKSEG1			0xa0000000
+#endif
+#ifndef CKSEG2
 #define CKSEG2			0xc0000000
+#endif
+#ifndef CKSEG3
 #define CKSEG3			0xe0000000
+#endif
 
 #endif
 
diff --git a/arch/mips/include/asm/mach-generic/spaces.h b/arch/mips/include/asm/mach-generic/spaces.h
index c3ac06a6acd2..623ae21d4182 100644
--- a/arch/mips/include/asm/mach-generic/spaces.h
+++ b/arch/mips/include/asm/mach-generic/spaces.h
@@ -30,11 +30,13 @@
 #endif /* __ASSEMBLY__ */
 
 #ifdef CONFIG_32BIT
+#ifndef CAC_BASE
 #ifdef CONFIG_KVM_GUEST
 #define CAC_BASE		_AC(0x40000000, UL)
 #else
 #define CAC_BASE		_AC(0x80000000, UL)
 #endif
+#endif
 #ifndef IO_BASE
 #define IO_BASE			_AC(0xa0000000, UL)
 #endif
diff --git a/arch/mips/include/asm/pgtable-32.h b/arch/mips/include/asm/pgtable-32.h
index 6c0532d7b211..613fd57edfa3 100644
--- a/arch/mips/include/asm/pgtable-32.h
+++ b/arch/mips/include/asm/pgtable-32.h
@@ -100,11 +100,13 @@ extern int add_temporary_entry(unsigned long entrylo0, unsigned long entrylo1,
 #define PKMAP_END	((FIXADDR_START) & ~((LAST_PKMAP << PAGE_SHIFT)-1))
 #define PKMAP_BASE	(PKMAP_END - PAGE_SIZE * LAST_PKMAP)
 
+#ifndef VMALLOC_END
 #ifdef CONFIG_HIGHMEM
 # define VMALLOC_END	(PKMAP_BASE-2*PAGE_SIZE)
 #else
 # define VMALLOC_END	(FIXADDR_START-2*PAGE_SIZE)
 #endif
+#endif
 
 #ifdef CONFIG_PHYS_ADDR_T_64BIT
 #define pte_ERROR(e) \
-- 
2.20.1

